File: CWM.Model/Models/Searches/WorkOrderSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Core.Models.Searches
{
    public class WorkOrderSearch : BaseSearch
    {
        public string? Name { get; set; }

        public bool IncludeVehicle { get; set; }
        public bool IncludePayment { get; set; }

        public int? AppointmentId { get; set; }
        public string? EmployeeUsername { get; set; }
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/searches/work_order_search.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search.dart';

part 'work_order_search.g.dart';

@JsonSerializable()
class WorkOrderSearch extends BaseSearch {
  String? name;
  bool includeVehicle = false;
  bool includePayment = false;
  int? appointmentId;
  String? employeeUsername;
  int? userId;

  WorkOrderSearch();

  factory WorkOrderSearch.fromJson(Map<String, dynamic> json) =>
      _$WorkOrderSearchFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$WorkOrderSearchToJson(this);
}
